File: backend/src/main/java/de/keiss/selfhealing/backend/model/Connection.java
package de.keiss.selfhealing.backend.model;

import java.time.LocalTime;
import java.util.List;

public record Connection(String id, String from, String to, LocalTime departure, LocalTime arrival, int durationMinutes,
        int transfers, double priceEuro, List<String> trainTypes, List<Leg> legs) {

    public record Leg(String from, String to, LocalTime departure, LocalTime arrival, String trainType,
            String trainNumber, String platform) {
    }
}

File: android-app/app/src/main/java/de/keiss/selfhealing/app/data/Connection.kt
package de.keiss.selfhealing.app.data

data class Connection(
    val id: String,
    val from: String,
    val to: String,
    val departure: String,
    val arrival: String,
    val durationMinutes: Int,
    val transfers: Int,
    val priceEuro: Double,
    val trainTypes: List<String>,
    val legs: List<Leg>
)

data class Leg(
    val from: String,
    val to: String,
    val departure: String,
    val arrival: String,
    val trainType: String,
    val trainNumber: String,
    val platform: String
)
